### DISCUSSION TITLE
A couple recent ATtiny 1626 projects

### DISCUSSION THREAD

**Discussion by jvasileff** [SpenceKonde/megaTinyCore#627] [Show and tell]
Thanks to Spence and megaTinyCore for helping fuel my electronics hobby. I thought I'd share a couple projects I recently got working that use megaTinyCore and ATtiny 1626 MCUs.

The first is a tiny voltmeter that's actually accurate enough to be usefull - about +-0.2% and +-2mV over its range 0V to 30V.

The second sits along side a Raspberry Pi to help manage my Prusa MK3s 3d printer. It currently handles PWM for 24V overhead LED strip lights, PWM and constant-current for a camera ring light, and on/off power management for the Raspberry Pi. It communicates with the PI via i2c, serial, and UPDI, as well as the Pi's run and global_en power management pins. I plan to add a camera trigger for time lapses, enclosure temperature monitoring, and who knows what else.

![tiny-voltmeter-1](https://user-images.githubusercontent.com/236050/150839046-9e2fc93a-6e51-4da3-9307-68c6464f6c03.jpeg)

![tiny-voltmeter-2](https://user-images.githubusercontent.com/236050/150839060-93726b51-54fd-4b06-92ed-4e51d213f693.jpeg)

![prusa-pi-1](https://user-images.githubusercontent.com/236050/150839093-e41ae4f4-f138-433d-848f-8a4659e1edcd.jpeg)

![prusa-pi-2](https://user-images.githubusercontent.com/236050/150839182-a23e9d63-b815-4b16-a2ff-f44beb4c37ba.jpeg)

**Comment by dapeake2** [comment]
Excellent work on the voltmeter project. The miniaturization is great. Perfect platform for a logic probe / volt meter, with a fine needle probe, and, as part of the probe handle, a 7mm seven segment display (smallest I've ever seen) as a means to keep the probe tip and the numeric readout in view without having to swivel ones head to view a voltmeter between readings.

Dan

**Reply by jvasileff** [reply]
That's a great idea, I may give it a try. I don't think it would be too hard to 3d print something that would morph the box shape of the display into a cone or cylinder with a cavity for the wire and tip. I suppose the diameter of a 3d printed probe may be a bit large, but it would probably be ok if the tip (perhaps a thumb tack) extended out a couple mm.

**Comment by dapeake2** [comment]
I thought about making the device attach, by a slight interference fit, to any existing DVM probe using a flexible "C" shape that clips on like a stage mic.

Dan
